feature/s3/transfermanager/api_op_GetObject.go

```diff
@@ -0,0 +1,927 @@
+
+	input.ExpectedBucketOwner = nzstring(i.ExpectedBucketOwner)
+	input.IfMatch = nzstring(i.IfMatch)
+	input.IfNoneMatch = nzstring(i.IfNoneMatch)
+	input.IfModifiedSince = nztime(i.IfModifiedSince)
+	input.IfUnmodifiedSince = nztime(i.IfUnmodifiedSince)
+	input.ResponseCacheControl = nzstring(i.ResponseCacheControl)
+	input.ResponseContentDisposition = nzstring(i.ResponseContentDisposition)
+	input.ResponseContentEncoding = nzstring(i.ResponseContentEncoding)
+	input.ResponseContentLanguage = nzstring(i.ResponseContentLanguage)
+	input.ResponseContentType = nzstring(i.ResponseContentType)
+	input.ResponseExpires = nztime(i.ResponseExpires)
+	input.SSECustomerAlgorithm = nzstring(i.SSECustomerAlgorithm)
+	input.SSECustomerKey = nzstring(i.SSECustomerKey)
+	input.SSECustomerKeyMD5 = nzstring(i.SSECustomerKeyMD5)
+	input.VersionId = nzstring(i.VersionID)
+
+	return input
+}
+
+// GetObjectOutput represents a response from GetObject() or DownloadObject() call. It contains common fields
+// of s3 GetObject output
+type GetObjectOutput struct {
+	// Indicates that a range of bytes was specified in the request.
+	AcceptRanges string
+
+	// Object data.
+	Body io.ReadCloser
+
+	// Indicates whether the object uses an S3 Bucket Key for server-side encryption
+	// with Key Management Service (KMS) keys (SSE-KMS).
+	BucketKeyEnabled bool
+
+	// Specifies caching behavior along the request/reply chain.
+	CacheControl string
+
+	// Specifies if the response checksum validation is enabled
+	ChecksumMode types.ChecksumMode
+
+	// The base64-encoded, 32-bit CRC-32 checksum of the object. This will only be
+	// present if it was uploaded with the object. For more information, see [Checking object integrity]in the
+	// Amazon S3 User Guide.
+	//
+	// [Checking object integrity]: https://docs.aws.amazon.com/AmazonS3/latest/userguide/checking-object-integrity.html
+	ChecksumCRC32 string
+
+	// The base64-encoded, 32-bit CRC-32C checksum of the object. This will only be
+	// present if it was uploaded with the object. For more information, see [Checking object integrity]in the
+	// Amazon S3 User Guide.
+	//
+	// [Checking object integrity]: https://docs.aws.amazon.com/AmazonS3/latest/userguide/checking-object-integrity.html
+	ChecksumCRC32C string
+
+	// The base64-encoded, 160-bit SHA-1 digest of the object. This will only be
+	// present if it was uploaded with the object. For more information, see [Checking object integrity]in the
+	// Amazon S3 User Guide.
+	//
+	// [Checking object integrity]: https://docs.aws.amazon.com/AmazonS3/latest/userguide/checking-object-integrity.html
+	ChecksumSHA1 string
+
+	// The base64-encoded, 256-bit SHA-256 digest of the object. This will only be
+	// present if it was uploaded with the object. For more information, see [Checking object integrity]in the
+	// Amazon S3 User Guide.
+	//
+	// [Checking object integrity]: https://docs.aws.amazon.com/AmazonS3/latest/userguide/checking-object-integrity.html
+	ChecksumSHA256 string
```

Comment:
Are you actually verifying checksums in your GetObject implementation?

Reply:
Yes, the checksum validation is enabled by default unless user specify that in their input

Reply:
The validation skip log also presented when multi-part download an large object uploaded without checksum